fiber-unix/src/worker.ml

```diff
@@ -0,0 +1,32 @@
+open Import
+
+type 'a t = { work_chan : 'a Channel.t } [@@unboxed]
+
+type task = Channel.elt_in_channel
+
+let run t f =
+  let rec loop () =
+    match Channel.get t.work_chan with
+    | Ok v ->
+      f v;
```

Comment:
That's a good question. The previous implementation of `worker` did nothing if `f` raises; neither does this implementation. I think that's a correct behavior because we don't return the result of `f v`, so the only option I see is to swallow the exception which is obviously bad. I'll warn in the doc comment that `f` shouldn't raise unless the user wants the exception go wild.